test/tests/unit/connection_string_tests.js

```diff
@@ -100,11 +100,8 @@ describe('Connection String', function() {
       'mongodb://localhost/?compressors=zlib&zlibCompressionLevel=4',
       (err, result) => {
         expect(err).to.not.exist;
-        expect(result.options).to.have.property('compression');
-        expect(result.options.compression).to.eql({
-          compressors: ['zlib'],
-          zlibCompressionLevel: 4
-        });
+        expect(result.options.compressors).to.eql(['zlib']);
```

Comment:
Will this break anything in native/elsewhere? It looks like we used to verify that `compression` existed, and now we don't. Let's be careful about removing these types of checks

Reply:
Hmm it might since I'm removing L260. Should I support both `compressors` and `compression`?

Reply:
If this breaks compression we should be seeing it in unit tests. Like if this does what I was proposing it might, then compression won't work at all. Can you go take a look and verify that it's not doing that?

Reply:
No unit tests fail. I'm pretty sure this doesn't break compression because tests like [this](https://github.com/mongodb-js/mongodb-core/blob/master/test/tests/functional/server_tests.js#L978) pass.

Reply:
I wonder about code like [this](https://github.com/mongodb-js/mongodb-core/blob/master/lib/connection/connect.js#L105-L116), and maybe that we're only checking one path for compression maybe? That definitely looks to me like we're looking for `options.compression` not `options.compressors`

Reply:
Yeah, that code looks like it should break. I guess we're not testing that anywhere?

I can support both `compression` and `compressors` and file a ticket to move things from checking/using `compression` to `compressors`?

Reply:
iirc, `compression` was added to wrap `compressors`, and `zlibCompressionLevel`. I'm not sure whether or not we should remove it at this point, but it's probably still possible to pass the uri spec tests using `compression`. It would be great if you could actually write a test that covers what we found here today though.

Reply:
I added a test in the [native PR](https://github.com/mongodb/node-mongodb-native/pull/1929).